compiler/src/dotty/tools/dotc/util/SourcePosition.scala

```diff
@@ -26,11 +26,12 @@ extends interfaces.SourcePosition {
     source.content.slice(source.startOfLine(start), source.nextLine(end))
 
   /** The lines of the position */
-  def lines: List[Int] =
-    List.range(source.offsetToLine(start), source.offsetToLine(end + 1)) match {
-      case Nil => line :: Nil
-      case xs => xs
-    }
+  def lines: List[Int] = {
+    val startOffset = source.offsetToLine(start)
+    val endOffset = source.offsetToLine(end + 1)
+    if (startOffset >= endOffset) line :: Nil
+    else List.tabulate(endOffset - startOffset)(i => i + startOffset)
```

Comment:
Removing the offset to add it back is confusing, I would just keep the original code here since it shouldn't be performance-sensitive, or do `(startOffset to endOffset).toList` maybe.